File: src/iterative_sorting/iterative_sorting.py
import random


def selection_sort(arr):
    # loop through n-1 elements
    for i in range(0, len(arr) - 1):
        smallest_index = i

        # Find smallest element
        for cur_index in range(smallest_index + 1, len(arr)):
            if arr[cur_index] < arr[smallest_index]:
                smallest_index = cur_index

        # Swap
        arr.insert(i, arr.pop(smallest_index))

    return arr


# Test - Should print list with 0 - 9 ascending
print(selection_sort(random.sample(range(10), 10)))

# TO-DO:  implement the Bubble Sort function below
def bubble_sort(arr):

    return arr


# STRETCH: implement the Count Sort function below
def count_sort(arr, maximum=-1):

    return arr

File: src/iterative_sorting/iterative_sorting.js
//* Selection Sort
function selection_sort(arr) {
  let smallest_index;
  let temp;

  for (let i = 0; i < arr.length - 1; i++) {
    smallest_index = i;
    temp = arr[i]; // For swapping

    for (let cur_index = smallest_index + 1; cur_index < arr.length; cur_index++) {
      if (arr[smallest_index] > arr[cur_index]) {
        smallest_index = cur_index;
      }
    }

    //Swap elements
    arr[i] = arr[smallest_index];
    arr[smallest_index] = temp;
  }

  return arr;
}

console.log("Selection Sort : ", selection_sort([6, 1, 4, 5, 3, 0, 2]));

//* Bubble Sort
function bubble_sort(arr) {
  let didSwap = true;
  let temp;

  while (didSwap) {
    didSwap = false;
    for (let i = 0; i < arr.length - 1; i++) {
      temp = arr[i];
      if (arr[i] > arr[i + 1]) {
        arr[i] = arr[i + 1];
        arr[i + 1] = temp;
        didSwap = true;
      }
    }
  }

  return arr;
}

console.log("Bubble Sort: ", bubble_sort([6, 1, 4, 5, 3, 0, 2]));
